src/composables/storageRef.ts

```diff
@@ -34,13 +42,16 @@ export function useStorageRef<T = string | object | any[]>(
 ) {
   const {
     serializer,
+    isSecure = false,
     backgroundSync = false,
     migrations,
     onRestored,
   } = options;
 
   let watcherDisabled = false; // Avoid watcher going infinite loop
-  const state = ref(initialState) as Ref<T>; // https://github.com/vuejs/core/issues/2136
+  const state = ref(initialState) as Ref<T>; // https://github.com/vuejs/core/issues/2136/
+  // Select storage type
+  let storage = isSecure ? null : WalletStorage;
```

Comment:
I think this is a bad pattern to choose the storage provider here but then no matter if `isSecure` is set use the `WalletStorage` in further code (see line 72).

Reply:
On line 72 we are using `WalletStorage` **only** if `isSecure` is `true` in order to retrieve the stored (unencrypted) value